Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -17,6 +17,112 @@ import NIO
 let crlf: StaticString = "\r\n"
 let headerSeparator: StaticString = ": "
 
+fileprivate let comma = UInt8(",".utf8CString[0])
+fileprivate let whiteSpaces = [" ", "\t"]
+                              .map( {UInt8($0.utf8CString[0])} )
+
+/// An `IteratorProtocol` that can iterate through comma separated list of values for a certain
+/// header
+///
+///  **Example:**
+///
+/// Suppose you have this headers:
+///
+///      Connection: keep-alive, x-server
+///      Content-Type: text/html
+///      Connection: other
+///
+/// You can iterate using this struct on those headers, for values of `Connection`, to get
+/// `keep-alive`, then `x-server`, then `other`
+///
+public struct HTTPListHeaderIterator<T: Collection>: IteratorProtocol where T.Element == UInt8 {
+    
+    public typealias Element = ByteBufferView
+    
+    private var isNowOnAField = false
+    private var currentHeaderIndex: Int = -1
+    private var byteBuffer: ByteBuffer
+    private var singleValueViewIterator: Array<ByteBufferView>.Iterator
+    private let headerName: T
+    private let headers: [HTTPHeader]
+    
+    /// Returns next index in headers
+    ///
+    /// - Parameter current: The index to begin iteration at
+    /// - Returns: The next index of the header in header array, or `nil` if not found
+    internal func nextHeaderIndex(ifCurrentIs current: Int) -> Int? {
```

Comment:
For consistency with the standard library, let's call this `headerIndex(after:)`.